compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/partition_cut.rs

```diff
@@ -12,36 +12,98 @@ use crate::optimization::dag::solo_key::symbolic_variance::SymbolicVariance;
 const ROUND_INNER_MULTI_PARAMETER: bool = false;
 const ROUND_EXTERNAL_MULTI_PARAMETER: bool = !ROUND_INNER_MULTI_PARAMETER && true;
 
+#[derive(Hash, Eq, PartialEq, Clone, Debug)]
+pub struct ExternalPartition {
+    pub name: String,
+    // TODO add params (maybe just macros)
+}
```

Comment:
Do we need a string here ? If used for debugging, it indeed make sense, but maybe we could also have something equivalent for classic partitions ? Is that what you mean in your TODO below ?

Reply:
I don't think we have classic partitions here. Partitions will be coming from the compiler with a name as an identifier. And yes, it's helpful for debugging and visualizing the graph. The TODO is for parameters to be specified for the partition. Those should be coming from the compiler.

Reply:
And I was referring to todo line 32, sorry for not being clear.